src/main/java/uk/gov/gchq/palisade/resource/AbstractResource.java

```diff
@@ -16,59 +16,57 @@
 
 package uk.gov.gchq.palisade.resource;
 
-import org.apache.commons.lang3.builder.EqualsBuilder;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-
+import uk.gov.gchq.palisade.Generated;
 import uk.gov.gchq.palisade.ToStringBuilder;
 
 import java.util.Comparator;
 import java.util.Objects;
 
+import static java.util.Objects.requireNonNull;
+
 public abstract class AbstractResource implements Resource {
+
+    private static Comparator<Resource> comp = Comparator.comparing(Resource::getId);
     protected String id;
 
     public AbstractResource() {
     }
 
     public AbstractResource id(final String id) {
```

Comment:
This needs the @generated annotation on the mutator

Reply:
Done